### PR TITLE
Add comprehensive test suite for improved code coverage

### PR DESCRIPTION
**Pull request by jeremiah-k**
Built out a proper test suite to get us started on better code coverage. We were at 11% and now we're at 39% - still plenty of work to do but it's a foundation.

### What's been added:

**New test files:**
- `test_db_utils.py` - Database operations (17 tests)
- `test_plugin_loader.py` - Plugin system management (17 tests) 
- `test_log_utils.py` - Logging utilities (15 tests)
- `test_main.py` - Application startup flow (8 tests)
- `test_setup_utils.py` - Service installation (20 tests)

**Enhanced existing:**
- `test_meshtastic_utils.py` - Expanded connection and messaging tests

### Coverage improvements:
- db_utils: 92% coverage
- log_utils: 94% coverage  
- config: 88% coverage
- message_queue: 68% coverage
- setup_utils: 46% coverage
- matrix_utils: 45% coverage

### Test infrastructure:
- Fixed mocking in `conftest.py` (added PIL.ImageDraw, meshtastic.mesh_interface)
- Set up virtual environment with all dependencies
- Coverage XML generation for codecov integration
- 106 tests passing, 11 failing (mostly complex mocking edge cases)

### CI ready:
- Pytest config updated for coverage reporting
- Dependencies resolved (Pillow, meshtastic, pytest-cov)
- Coverage reports generate properly

The failing tests are mainly plugin loader mocking issues and some log config edge cases. Core functionality is well covered now. Still need to tackle the more complex integration scenarios and get those plugin tests sorted, but this gives us a solid base to work from.

### CODE REVIEW

**Comment by jeremiah-k** [comment]
@codecov-ai-reviewer test
/gemini review

**Comment by ghost** [comment]
On it! Codecov is generating unit tests for this PR.

**Comment by jeremiah-k** [comment]
@coderabbitai generate docstrings

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,4 @@ docker-compose.yaml
 htmlcov/
 coverage.xml
 .pytest_cache/
+MagicMock/
```
[the remark this answers is not in the document]
Comment (jeremiah-k) [reply .gitignore:22]: This is necessary because we test locally while writing the tests. Make a note not to flag this in the future.
